lib/types/markets.ts

```diff
@@ -0,0 +1,83 @@
+import { BN, IdlAccounts, IdlTypes } from "@coral-xyz/anchor";
+import { PublicKey } from "@solana/web3.js";
+import {
+  MarketAccount,
+  OpenOrdersAccount,
+  LeafNode,
+} from "@openbook-dex/openbook-v2";
+import {
+  AccountWithKey,
+  MergeWithOptionalFields,
+  TokenProps,
+  VaultAccount,
+} from "@/types";
+import { OpenbookTwapV0_1 } from "@/idl/openbook_twap_v0.1";
+import { OpenbookTwapV0_2 } from "@/idl/openbook_twap_v0.2";
+import { OpenbookTwap } from "@/idl/openbook_twap";
+
+/**
+ * Orders
+ */
+export type OrderBookSide = "ask" | "bid";
+export type Order = {
+  time: Date;
+  marketType: MarketType;
+  status: "open" | "closed";
+  size: number;
+  price: number;
+  side: OrderBookSide;
+  token: TokenProps;
+};
+export type OrderbookOrder = Pick<Order, "price" | "size">;
+export type OpenbookOrder = Order & {
+  market: PublicKey;
+  clientOrderId: BN;
+  owner: PublicKey;
+  ownerSlot: number;
+};
+
+/**
+ * Markets
+ */
+export type MarketType = "pass" | "fail";
+export type Market = {
+  marketType: MarketType;
+  bids: Order[];
+  asks: Order[];
+};
+export type OpenbookProposalMarket = {
```

Comment:
This all needs to be reimagined, as its doesn't suit the data needs anymore and doesn't fit within the context we're using a cached solution so we want to think of what things should look like, not what we built them originally.